src/test/groovy/ee/tuleva/onboarding/auth/jwt/JwtAuthorizationFilterSpec.groovy

```diff
@@ -0,0 +1,122 @@
+import io.jsonwebtoken.Jwts
+import org.springframework.mock.web.MockHttpServletRequest
+import org.springframework.mock.web.MockHttpServletResponse
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.context.SecurityContextHolder
+import spock.lang.Specification
+
+import javax.servlet.FilterChain
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneId
+
+import static ee.tuleva.onboarding.auth.AuthenticatedPersonFixture.sampleAuthenticatedPersonAndMember
+import static ee.tuleva.onboarding.auth.KeyStoreFixture.keyStore
+import static ee.tuleva.onboarding.auth.KeyStoreFixture.keyStorePassword
+import static java.time.temporal.ChronoUnit.HOURS
+
+class JwtAuthorizationFilterSpec extends Specification {
+
+  private final Clock clock = Clock.fixed(Instant.EPOCH, ZoneId.of("UTC"))
+
+  private final JwtTokenUtil jwtTokenUtil = new JwtTokenUtil(keyStore(), keyStorePassword, clock)
+
+  private final PrincipalService principalService = Mock()
+
+  private final JwtAuthorizationFilter filter = new JwtAuthorizationFilter(jwtTokenUtil, principalService)
+
+  def cleanup() {
+    SecurityContextHolder.clearContext()
+  }
+
+  def "does not do anything on empty request"() {
+    given:
+    def request = new MockHttpServletRequest()
+    def response = new MockHttpServletResponse()
+    def filterChain = Mock(FilterChain)
+    when:
+    filter.doFilterInternal(request, response, filterChain)
+    then:
+    SecurityContextHolder.context.getAuthentication() == null
+    1 * filterChain.doFilter(request, response)
+  }
+
+  def "reads user info from a valid authorization header"() {
+    given:
+    def token = Jwts.builder()
+        .setSubject("38510309519")
+        .signWith(KeyStoreFixture.keyPair.private)
+        .setExpiration(Date.from(clock.instant().plus(1, HOURS)))
+        .claim("firstName", "Peeter")
+        .claim("lastName", "Meeter")
+        .claim("authorities", new String[]{"USER"})
+        .compact()
+    def person = sampleAuthenticatedPersonAndMember()
+        .firstName("Peeter")
+        .lastName("Meeter")
+        .personalCode("38510309519")
+        .build()
+    def request = new MockHttpServletRequest()
+    request.addHeader("Authorization", "Bearer " + token)
+    def response = new MockHttpServletResponse()
+    def filterChain = Mock(FilterChain)
+    principalService.getFrom(_, _) >> person
+    when:
+    filter.doFilterInternal(request, response, filterChain)
+    then:
+    SecurityContextHolder.context.getAuthentication() != null
+    (SecurityContextHolder.context.getAuthentication().getPrincipal() as AuthenticatedPerson).firstName == "Peeter"
+    (SecurityContextHolder.context.getAuthentication().getPrincipal() as AuthenticatedPerson).lastName == "Meeter"
+    (SecurityContextHolder.context.getAuthentication().getPrincipal() as AuthenticatedPerson).personalCode == "38510309519"
```

Comment:
Maybe put `SecurityContextHolder.context.getAuthentication().getPrincipal() as AuthenticatedPerson` into a variable so we can only do it once. Makes the asserts more readable.
i.e. `authenticatedPerson.firstName == "Peeter"`